drivers/sensor/ina23x/ina230.c

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright 2021 The Chromium OS Authors
+ * Copyright 2021 Grinn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_ina230
+
+#include <logging/log.h>
+#include <drivers/sensor.h>
+#include "ina230.h"
+#include "ina23x_common.h"
+
+LOG_MODULE_REGISTER(INA230, CONFIG_SENSOR_LOG_LEVEL);
+
+/**
+ * @brief Macro for creating the INA230 calibration value
+ *        CALIB = (5120 / (current_lsb * rshunt))
+ *        NOTE: The 5120 value is a fixed value internal to the
+ *              INA230 that is used to ensure scaling is properly
+ *              maintained.
+ *
+ * @param current_lsb Value of the Current register LSB in milliamps.
+ * @param rshunt Shunt resistor value in milliohms.
+ */
+ #define INA230_CALIB(current_lsb, rshunt) (5120 / ((current_lsb) * (rshunt)))
+
+/**
+ * @brief Macro to convert raw Bus voltage to millivolts when current_lsb is
+ *        set to 1mA.
+ *
+ * @param reg value read from bus voltage register.
+ */
+#define INA230_BUS_MV(reg) ((reg) * 125 / 100)
+
+/**
+ * @brief Macro to convert raw power value to milliwatts when current_lsb is
+ *        set to 1mA.
+ *
+ * @param reg value read from power register.
+ */
+#define INA230_POW_MW(reg) ((reg) * 25)
```

Comment:
I'd prefer not to hide these arithmetic operations behind macros, especially since they're only being used once

Reply:
In my opinion it's better to have it as a macro with an explanation since it's more clearer for the user and they are being used twice per channel get request.

Reply:
`INA230_CALIB` is only used once.

`INA230_BUS_MV` and `INA230_POW_MW` are each used twice, but within the same function scope. It would be more clear to use an intermediate local variable with the explanation there, and it will also be easier to see if any necessary typecasting is handled correctly.

Reply:
Macros replaced.